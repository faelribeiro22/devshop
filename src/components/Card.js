import React, { Component } from 'react'

export default class Card extends Component {

  render() {
    const profile = this.props;
    return (
      <div className="col-md-4">
        <div className="card mb-4 shadow-sm">
          <img
            className="card-img-top card-photo"
            src={profile.profileAvatar}
            alt="Avatar"
          />
          <div className="card-body">
            <label>{profile.profileName}</label>
          </div>
        </div>
      </div>
    )
  }
}
