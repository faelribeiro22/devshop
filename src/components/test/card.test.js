import React from 'react';
import Card from '../Card';
import renderer from 'react-test-renderer';

const profileDefault = {
  nome: 'faelribeiro22',
  avatar: 'https://avatars1.githubusercontent.com/u/10289821?v=4',
}

it('Test card component renderer', () => {
  const component = renderer.create(
    <Card profile={profileDefault} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

