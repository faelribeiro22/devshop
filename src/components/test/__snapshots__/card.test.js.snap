// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test card component renderer 1`] = `
<div
  className="col-md-4"
>
  <div
    className="card mb-4 shadow-sm"
  >
    <img
      alt="Avatar"
      className="card-img-top card-photo"
    />
    <div
      className="card-body"
    >
      <label />
    </div>
  </div>
</div>
`;
